validate.go

```diff
@@ -24,6 +24,12 @@ func (validator *Validator) ValidateWithContext(ctx context.Context, subject int
 	var violations ViolationList
 	var err error
 
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic recovered in ValidateWithContext: %v", r)
```

Comment:
If you want to return the error here, you would need to use the named returned values:
```
func (validator *Validator) ValidateWithContext(ctx context.Context, subject interface{}) (isValid bool, list ViolationList, finalErr error) {
 defer func() {
		if r := recover(); r != nil {
			finalErr = fmt.Errorf("panic recovered in ValidateWithContext: %v", r)
       }
   }
}
```

Reply:
the return statements is happening here:

```
 if err != nil {
        // If err is set, return immediately
        return false, nil, err
    }
```